components/http/okHttp/src/main/java/com/microsoft/kiota/http/OkHttpRequestAdapter.java

```diff
@@ -899,6 +900,30 @@ public MediaType contentType() {
                                     }
                                 }
 
+                                @Override
+                                public long contentLength() {
+                                    long length;
+                                    final Set<String> contentLength =
+                                            requestInfo.headers.getOrDefault(
+                                                    contentLengthHeaderKey, new HashSet<>());
+                                    if (contentLength.isEmpty()
+                                            && requestInfo.content
+                                                    instanceof ByteArrayInputStream) {
+                                        final ByteArrayInputStream contentStream =
+                                                (ByteArrayInputStream) requestInfo.content;
+                                        length = contentStream.available();
```

Comment:
I'm so sorry I have not followed up on this subject, but, as noticed here: https://github.com/microsoft/kiota-java/pull/1088#discussion_r1494670806

`available` is *NOT* the API you need to use in this context since it has very very different semantics from the expected behavior.
Have you explored different solutions?
Do we have a reproducer of the original issue that triggered those changes?

Reply:
Unfortunately, due to the support and semantics, I'm using workarounds in the Apicurio Registry testsuite to properly perform file uploads.
But I'm looking forward to a real resolution of the issue.

If you need me to chip in I would love a minimal repoducer, maybe @araneolus would be able to provide one 🙏